<?xml version="1.0" ?>
<!-- TODO: Add GPS sensor -->
<robot name="lexus_rx_450h">
  <material name="black">
    <color rgba="0.2 0.2 0.2 1"/>
  </material>
  <material name="blue">
    <color rgba="0.1 0.1 0.5 1"/>
  </material>

  <!-- base_link relates to its projection on the ground by this offset:
    <origin xyz="0 0 0.31" rpy="0 0 0"/>
  -->


  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://lexus_rx_450h_description/meshes/LexusRX.obj" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black"/>
      <origin xyz="0.83 0 0.31" rpy="0 0 1.570796"/>
    </visual>
    <collision>
      <geometry>
        <box size="5.5 2.25 2"/>
      </geometry>
      <origin xyz="0.83 0 0.31" rpy="0 0 0"/>
    </collision>
    <inertial>
      <!-- TODO: Get or calculate inertia values -->
      <origin xyz="0 0 0.013" rpy="0 0 0"/>
      <mass value="0.25"/>
      <inertia ixx="0.0003350833" ixy="0" ixz="0"
        iyy="0.000560417" iyz="0"
        izz="0.000478833"/>
    </inertial>
  </link>

  <link name="lidar_front">
    <visual>
      <geometry>
        <cylinder length="0.0869" radius="0.05165"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0869" radius="0.05165"/>
      </geometry>
    </collision>
    <inertial>
      <!-- TODO: Get or calculate inertia values -->
      <mass value="0.1"/>
      <inertia ixx="1.02083e-5" ixy="0" ixz="0"
        iyy="1.02083e-5" iyz="0"
        izz="2e-5"/>
    </inertial>
  </link>

  <joint name="lidar_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_front"/>
    <origin xyz="1.4 0 1.422" rpy="0.0215759 0.125004 0.111496"/>
  </joint>

  <link name="lidar_rear">
    <visual>
      <geometry>
        <cylinder length="0.0869" radius="0.05165"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0869" radius="0.05165"/>
      </geometry>
    </collision>
    <inertial>
      <!-- TODO: Get or calculate inertia values -->
      <mass value="0.1"/>
      <inertia ixx="1.02083e-5" ixy="0" ixz="0"
        iyy="1.02083e-5" iyz="0"
        izz="2e-5"/>
    </inertial>
  </link>

  <joint name="lidar_rear_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_rear"/>
    <origin xyz="0.1 0.01 1.422" rpy="0 0.146539 3.13"/>
  </joint>
</robot>
